### PR TITLE
feat(update): Allow to use a bot account for the `update` action

### PR DESCRIPTION
**Pull request by ErwanSimonetti**
The `update` action should be usable with a bot account

Fixes MRGFY-972

### CODE REVIEW

File: mergify_engine/branch_updater.py [hunk 1]
```diff
@@ -184,13 +184,14 @@ async def _do_rebase(
         await git.cleanup()
 
 
-async def update_with_api(ctxt: context.Context) -> None:
+async def update_with_api(ctxt: context.Context, github_user) -> None:
```
Comment (Syffe) [inline mergify_engine/branch_updater.py:187]: missing typing for `github_user` parameter

File: mergify_engine/tests/functional/actions/test_update.py [hunk 2]
```diff
@@ -57,6 +57,42 @@ async def test_update_action(self):
         assert commits[-1]["commit"]["author"]["name"] == config.BOT_USER_LOGIN
         assert commits[-1]["commit"]["message"].startswith("Merge branch")
 
+    async def test_update_action_with_bot(self):
```
Comment (Syffe) [inline mergify_engine/tests/functional/actions/test_update.py:60]:
```suggestion
    async def test_update_action_with_bot(self) -> None:
```

**Review by sileht — CHANGES_REQUESTED** [review]
I have the feeling that the zfixtures should only update TestUpdateAction* since you didn't touch other files.

File: docs/source/actions/update.rst [hunk 3]
```diff
@@ -12,6 +12,8 @@ update
 The ``update`` action updates the pull request against its base branch. It
 works by merging the base branch into the head branch of the pull request.
 
+It is possible to use this action with a bot.
```
Comment (sileht) [inline docs/source/actions/update.rst:15]: You must add a rst table like other actions.

File: releasenotes/notes/bot-user-update-action-6af260955bd77a61.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    The ``update`` action is now usable with a bot
```
Comment (sileht) [inline releasenotes/notes/bot-user-update-action-6af260955bd77a61.yaml:4]:
```suggestion
    The ``update`` action gets a new option ``bot_account``.

File: mergify_engine/actions/update.py [hunk 5]
```diff
@@ -46,12 +77,9 @@ async def run(ctxt: context.Context, rule: rules.EvaluatedRule) -> check_api.Res
             )
         else:
             await signals.send(
-                ctxt.repository,
-                ctxt.pull["number"],
-                "action.update",
-                signals.EventNoMetadata(),
-                rule.get_signal_trigger(),
+                ctxt, "action.rebase", {"bot_account": bool(self.config["bot_account"])}
```
Comment (Greesb) [inline mergify_engine/actions/update.py:80]: Changing from `update` to `rebase` ? Aren't the `update` and `rebase` 2 different behaviors ?

File: mergify_engine/actions/update.py [hunk 6]
```diff
@@ -32,10 +38,35 @@ class UpdateAction(actions.Action):
...
+            return check_api.Result(e.status, e.title, e.reason)
+
+        github_user: typing.Optional[user_tokens.UserTokensUser] = None
+        if bot_account:
+            tokens = await ctxt.repository.installation.get_user_tokens()
+            github_user = tokens.get_token_for(bot_account)
+            if not github_user:
+                return check_api.Result(
+                    check_api.Conclusion.FAILURE,
+                    f"Unable to comment: user `{bot_account}` is unknown. ",
```
Comment (Greesb) [inline mergify_engine/actions/update.py:66]: Shouldn't it be
```suggestion
                    f"Unable to update: user `{bot_account}` is unknown. ",
```
?

File: mergify_engine/actions/update.py [hunk 7]
```diff
@@ -32,10 +38,35 @@ class UpdateAction(actions.Action):
...
+    async def run(
+        self, ctxt: context.Context, rule: rules.EvaluatedRule
+    ) -> check_api.Result:
+        try:
+            bot_account = await action_utils.render_bot_account(
+                ctxt,
+                self.config["bot_account"],
+                option_name="bot_account",
+                required_feature=subscription.Features.BOT_ACCOUNT,
+                missing_feature_message="Rebase with `update_bot_account` set is unavailable",
```
Comment (Greesb) [inline mergify_engine/actions/update.py:54]:
```suggestion
                missing_feature_message="Update with `update_bot_account` set is unavailable",
```

**Review by Greesb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**